src/core/include/mp-units/bits/scaling.h

```diff
@@ -0,0 +1,238 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2018 Mateusz Pusz
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#pragma once
+
+// IWYU pragma: private, include <mp-units/framework.h>
+#include <mp-units/bits/fixed_point.h>
+#include <mp-units/framework/customization_points.h>
+
+#ifndef MP_UNITS_IN_MODULE_INTERFACE
+#ifdef MP_UNITS_IMPORT_STD
+import std;
+#else
+#include <concepts>
+#endif
+#endif
+
+
+namespace mp_units {
+
+namespace detail {
+
+template<std::floating_point A, std::floating_point B>
+using minimal_floating_point_type =
+  std::conditional_t<(std::numeric_limits<A>::digits >= std::numeric_limits<B>::digits), A, B>;
+
+template<typename To, typename T>
+constexpr auto cast_integral(const T& value)
+{
+  if constexpr (std::is_integral_v<T>) {
+    return static_cast<To>(value);
+  } else {
+    return value;
```

Comment:
Not really. The specification of `cast_integral<To>(const From)&` (an implementation detail) is *"cast to `To`, if `From` is integral"*. The point is that the default implementation of `scale` basically has two separate cases:
 - "use `operator*(float)`": Applied whenever at least one of `From` and `To` is `treat_as_floating_point`. 
 - "use a fixed-point multiplication": Can only be directly be applied to standard integral types, so is used just for those, plus custom types which are implicitly convertible and have an integral `value_type`.
 
So the first case is much broader. For standard types, this in particular includes the case where `From` is integer and `To` is floating-point. There, just multiplying the integer with some floating-point type may cause precision-loss warnigs to emitted. Those are "bogus", because whenever `To` is explicitly specified, we're in fact in an "explicit cast" path. So we need to do an explicit cast here. But on the other hand, we cannot unilaterally do an explicit cast, because many types that should go through that path are not at all convertible to that floating-point type (e.g. `std::complex`).

There are other ways to handle this. We could also just silence the warning locally, or branch-out the mixed int/float paths further above.